### PR TITLE
chore: pass session manager into onFinish

### PR DESCRIPTION
**Pull request by pauldambra**
## Problem

It broke my brain thinking about what object reference would be closed over into this function

## Changes

Pass it in instead

## How did you test this code?

🙈

### CODE REVIEW

File: plugin-server/src/main/ingestion-queues/session-recording/session-recordings-blob-consumer.ts [hunk 1]
```diff
@@ -97,11 +97,11 @@ export class SessionRecordingBlobIngester {
...
                 topic,
-                (offsets) => {
+                (offsets, self: SessionManager) => {
                     this.offsetManager?.removeOffsets(topic, partition, offsets)
 
                     // If the SessionManager is done (flushed and with no more queued events) then we remove it to free up memory
-                    if (sessionManager.isEmpty) {
+                    if (self.isEmpty) {
                         this.sessions.delete(key)
                     }
```
Comment (benjackwhite) [inline plugin-server/src/main/ingestion-queues/session-recording/session-recordings-blob-consumer.ts:106]: Alternative: Why don't we clean this up in the periodic flush checker? It's only an optimisation to make sure we don't hold a load of empty objects in memory

**Review by benjackwhite — CHANGES_REQUESTED** [review]

File: plugin-server/src/main/ingestion-queues/session-recording/session-recordings-blob-consumer.ts (hunk 1 above)
Comment (pauldambra) [reply plugin-server/src/main/ingestion-queues/session-recording/session-recordings-blob-consumer.ts:106]: Way nicer 👨‍🍳 👌

**Review by benjackwhite — APPROVED** [review]

File: plugin-server/src/main/ingestion-queues/session-recording/session-recordings-blob-consumer.ts [hunk 2]
```diff
@@ -380,6 +376,15 @@ export class SessionRecordingBlobIngester {
...
                 })
+
+                if (sessionManager.isEmpty) {
+                    emptySessions.push(key)
+                }
+            })
+
+            emptySessions.forEach((key) => {
+                // If the SessionManager is done (flushed and with no more queued events) then we remove it to free up memory
+                this.sessions.delete(key)
```
[the remark this answers is not in the document]
Comment (pauldambra) [reply plugin-server/src/main/ingestion-queues/session-recording/session-recordings-blob-consumer.ts:387]: 🤯

**Comment by pauldambra** [comment]
things changed so much it's easier to do this from fresh

**Pull request closed without merging**